wgpu-hal/examples/raw-gles.rs

```diff
@@ -1,15 +1,16 @@
 //! This example shows interop with raw GLES contexts -
 //! the ability to hook up wgpu-hal to an existing context and draw into it.
+//!
+//! Emscripten build:
+//! 1. install emsdk
```

Comment:
It's not difficult, e.g. we can install emsdk and build raw-gles. Then we can run test in browser take screenshot, and compare it with expected, or something similar. I think you already should have last part for wasm32-unknown-unknown, and I can reuse it. So if you think it's worth to doing I will do.
 
Let me explain why I think it's not worth to doing. The emscripten build is testing same path that works on linux pc, so I use only khronos-egl crate in raw-gles code, probably we can use the same code for native integration. The only one difference is in window creation:

```rust
    let surface = unsafe {
        let window = std::ptr::null_mut::<std::ffi::c_void>();
        egl.create_window_surface(display, config, window, None)
    }
    .expect("unable to create surface");
```

We used null_ptr because it's allowed for em, if we put actual code of window creation for linux this example also should works in linux. Moreover this code path is same that egl.rs did internally. I mean there is no browser related stuff in emscripten integration.

The whole project is compiled by rust with emscripten triple, and when code is compiled emcc linker is used to produce WebAssemlby module. I want to say while code compiles it's not an wgpu/rust problem if something not working, probably it's emscripten bug, however egl implementation of em is very solid it didnt' change for years. I mean that if we do this in CI it will be more like integration test, that test compatibility of emscripten version against khronos-egl crate/rust/wgpu. 
Regarding native path - it's already covered by other tests. There are small amount of code  related to emscripten that are not covered, but I think that compilation check is enough here.

So, maybe is better to just add to README.md how to build example with emscripten, and who wants will check if it works.

Anyway, if you think it worth to add in CI, lets do it.

Reply:
Ok, that sounds rather convincing. Let's proceed with your original plan:)